### ISSUE TITLE
Does Paddle implement the SOC (sub-objectives consistency) module when reproducing MODNet?

### ISSUE THREAD

**Issue by unihornWwan** [PaddlePaddle/PaddleSeg#2945]
### 问题确认 Search before asking

- [X] 我已经搜索过问题，但是没有找到解答。I have searched the question and found no related answer.

### 请提出你的问题 Please ask your question

As mentioned in the original MODNet,  the function of SOC is to overfit the training set and perform poorly on real-world data, but I did not find the relevant part in the code, so are the current open-source MODNet parameter files of Paddle only performed supervised training using Distinction-646 and Adobe Deep Matting Dataset？Thanks.

**Comment by wuyefeilin** [comment]
we don't use the self-supervised SOC strategy. But when supervised training, we use the Lcons loss.

**Comment by unihornWwan** [comment]
> 

I see, thanks for your reply ！And I would like to ask if the developer will consider implementing this module in the future, thank you.

**Comment by wuyefeilin** [comment]
We will not implement this module in the future. If you want to use, only the loss module need to be modified.

**Comment by unihornWwan** [comment]
> We will not implement this module in the future. If you want to use, only the loss module need to be modified.

Thank you for your reply, but can you please describe in more detail? 
My view is that SOC training is an independent stage， in addition to using a dedicated unlabeled data loader, we also need to copy a converged model M to get M' to calculate Ldd Loss . So I think SOC needs an independent trainer to complete these, may I ask how the "only need to modify" loss you mentioned is implemented?

**Comment by wuyefeilin** [comment]
The loss is describe in this : https://github.com/PaddlePaddle/PaddleSeg/blob/release/2.7/Matting/ppmatting/models/modnet.py#L65

Maybe you need to write an another function to get the gt by M' for Ldd loss.
